### PR TITLE
Simulation userguide

### PR DESCRIPTION
**Pull request by AVHopp**
This PR introduces a user guide for BayBE's simulation capabilities. It briefly introduces the core concepts like `lookup` and the different kinds of simulation available.

Note that this PR was not designed to be perfect and final. Since there might or might not be an upcoming refactoring of the simulation capabilities, the goal of this PR was to have *some reasonable* user guide, not a perfect one.

In addition, I realized that there was no example using the `simulate_transfer_learning` function so far. I consequently added one example and called it "Cross Task Learning" as this was the best description I could find. The example is heavily based on the other Transfer Learning example.

### CODE REVIEW

File: docs/userguide/simulation.md [hunk 1]
```diff
@@ -1,4 +1,121 @@
...
+For a wide variety of applications of this functionality, we refer to the corresponding [examples](../../examples/Backtesting/Backtesting).
+
+## Terminology: What do we mean by "Simulation"?
+
+The term "simulation" can have two slightly different interpretations, depending on the applied context.
+
+1. It can refer to "backtesting" a particular experimental campaign on a fixed finite dataset.
+In this context, "simulation" means investigating what experimental trajectory we would have observed if we had applied the recommender in a certain defined context and restricted the possible parameter configurations to those contained in the dataset.
+
+2. It can refer to the simulation of an *actual* DOE loop, i.e., recommending experiments and retrieving the corresponding measurements, where the loop closure is realized in the form of a callable (black-box) function that can be queried during the optimization to provide target values.
```
Comment (Scienfitz) [inline docs/userguide/simulation.md:14]: add
`Such callable could for instance be a simple analytical function or numerical solver of a set of differential equations that describe a physical system.` (or similar)

File: docs/userguide/simulation.md [hunk 2]
```diff
@@ -1,4 +1,121 @@
...
+
+In BayBE, the simulation submodule allows a wide range of use cases and can even be used for "oracle predictions".
```
Comment (Scienfitz) [inline docs/userguide/simulation.md:16]: feels like this last paragraph should already be written below the next subheading `The lookup functionality`

File: docs/userguide/simulation.md [hunk 3]
```diff
@@ -1,4 +1,121 @@
...
+This is enabled by the proper use of the `lookup` functionality, which allows to either use fixed data sets, analytical functions, and general callbacks for retrieving target function values. 
+
+## The lookup functionality
+
+All functions require a `lookup` which is used to close the loop and return target values for points in the search space.
+It can be provided in the form of a dataframe or a `Callable`.
+
+```{note}
+Technically, the `lookup` can also be `None`. This results in the simulation producing fake results which is not useful for real applications and thus not discussed further.
```
Comment (Scienfitz) [inline docs/userguide/simulation.md:25]: `not useful` is maybe a bit to negatively worded here, it is for instance useful for simply testing code before you care about implementing the lookup. maybe just remove that subsentence so it says plain `fake results, which is not discussed further.` or beautify it with HIC

also prob replace `fake` by `random`

File: docs/userguide/simulation.md [hunk 4]
```diff
@@ -1,4 +1,121 @@
...
+```
+
+### Using a dataframe
+
+When choosing a dataframe, the dataframe needs to contain experimental settings and their target results.
```
Comment (Scienfitz) [inline docs/userguide/simulation.md:30]:
```suggestion
When choosing a dataframe, it needs to contain parameter combinations and their target results.
```

**Review by Scienfitz — APPROVED** [review]
i think the user guide is great and more than enough for the hackathon
not sure about the example

File: docs/userguide/simulation.md [hunk 5]
```diff
@@ -1,4 +1,121 @@
...
+An important aspect to keep in mind when providing a dataframe is that one either needs to ensure that **every** possible combination of parameters is part of the dataframe or that it can happen that a recommendation is made for which the target value cannot be found in the dataframe.
```
[the remark this answers is not in the document]
Comment (AVHopp) [reply docs/userguide/simulation.md]: Rewrote, please resolve if gucci.

File: docs/userguide/simulation.md (hunk 2 above)
Comment (AVHopp) [reply docs/userguide/simulation.md:16]: Which paragraph exactly do you mean here? The one starting with "In BayBE"?

File: docs/userguide/simulation.md [hunk 6]
```diff
@@ -1,4 +1,121 @@
...
+)
+~~~
+
+This function returns a dataframe that contains the results. For details on the columns of this dataframe as well as the dataframes returned by the other functions discussed here, we refer to the documentation of the submodule [here](baybe.simulation).
+
+## Simulating multiple scenarios
+
+The function [`simulate_scenarios`](baybe.simulation.simulate_scenarios) allows to specify multiple simulation settings at once.
+Instead of a single campaign, this function expects a dictionary of campaigns, mapping scenario identifiers to `Campaign` objects.
+In addition to the keyword arguments available for `simulate_experiment`, this function has the additional keyword argument `n_mc_iterations`. This keyword argument can be used to perform multiple Monte Carlo runs with  single call.
```
[the remark this answers is not in the document]
Comment (AVHopp) [reply docs/userguide/simulation.md]: Fixed the formulation, please resolve if gucci.

File: examples/Transfer_Learning/cross_task_learning.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+path = Path(sys.path[0])
+ax = sns.lineplot(
+    data=results,
+    marker="o",
+    markersize=10,
+    x="Num_Experiments",
+    y="Target_CumBest",
+    hue="Opt. Function",
+)
+create_example_plots(
```
[the remark this answers is not in the document]
Comment (AVHopp) [reply examples/Transfer_Learning/cross_task_learning.py:147]: I will put in the baselines manually and see how I feel about that. Will then probably share the plot here.

[the remark this answers is not in the document]
Comment (AVHopp) [reply examples/Transfer_Learning/cross_task_learning.py:147]: image
It is a bit unintuitive to use the negated version and the original version once and to minimize both, I I'd say we can do it that way.

File: docs/userguide/simulation.md (hunk 2 above)
Comment (Scienfitz) [reply docs/userguide/simulation.md:16]: yes, the last paragraph before the mentioned heading

Comment (AVHopp) [reply docs/userguide/simulation.md:18]: Changed.

**Pull request merged**